File: server/src/services/theme_package_service.ts
import { PrismaClient } from '@prisma/client';
import { BusinessError } from '@/middlewares/error_middleware';

const prisma = new PrismaClient();

interface ThemePackageItem {
    id: number;
    name: string;
    description?: string;
    coverUrl?: string;
    quoteIds: string;
    isMember: number;
    adUnlockDays: number;
    expireDate?: string;
}

async function list(): Promise<ThemePackageItem[]> {
    const items = await prisma.themePackage.findMany({
        where: { status: 1 },
        orderBy: { id: 'asc' },
    });

    return items.map((i) => ({
        id: i.id,
        name: i.name,
        description: i.description ?? undefined,
        coverUrl: i.coverUrl ?? undefined,
        quoteIds: i.quoteIds,
        isMember: i.isMember,
        adUnlockDays: i.adUnlockDays,
        expireDate: i.expireDate?.toISOString(),
    }));
}

async function getById(id: number): Promise<ThemePackageItem | null> {
    const item = await prisma.themePackage.findUnique({ where: { id } });
    if (!item || item.status !== 1) return null;

    return {
        id: item.id,
        name: item.name,
        description: item.description ?? undefined,
        coverUrl: item.coverUrl ?? undefined,
        quoteIds: item.quoteIds,
        isMember: item.isMember,
        adUnlockDays: item.adUnlockDays,
        expireDate: item.expireDate?.toISOString(),
    };
}

export { list, getById };

File: server-py/app/services/theme_package_service.py
"""主题包服务（对应 TS theme_package_service.ts）"""
from sqlalchemy.orm import Session

from ..models import ThemePackage


def _toItem(i: ThemePackage) -> dict:
    return {
        "id": i.id,
        "name": i.name,
        "description": i.description,
        "coverUrl": i.coverUrl,
        "quoteIds": i.quoteIds,
        "isMember": i.isMember,
        "adUnlockDays": i.adUnlockDays,
        "expireDate": i.expireDate.isoformat() if i.expireDate else None,
    }


def list(db: Session) -> list:
    items = (
        db.query(ThemePackage)
        .filter(ThemePackage.status == 1)
        .order_by(ThemePackage.id.asc())
        .all()
    )
    return [_toItem(i) for i in items]


def getById(db: Session, id: int) -> dict | None:
    item = db.query(ThemePackage).filter(ThemePackage.id == id).first()
    if not item or item.status != 1:
        return None
    return _toItem(item)
